### PR TITLE
DALA-715: fix the docker volumes

### PR DESCRIPTION
**Pull request by kmachura-mika**
DALA-715: fix the docker volumes - PR
Solution to avoid getting unnamed volumes in the Docker.
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [ ] At least one E2E Test exists testing the new feature
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed to a dev server using this branch
  - [x] It's verified that this version actually is the one deployed (check actuator/info for branch name and commit id!)
  - [x] The new feature is manually used/tested/observed on dev server
  - [x] All implemented Social Logins have been tested manually in the UI
- [x] If any work on the UI is to be merged, those changes were also documented in the Figma 
- [x] The local Dev stack still works: execute `startDevelopmentStack.sh`, execute npm run cypress and run Cypress Tests locally

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -147,16 +147,14 @@ services:
...
       - internal
     expose:
       - "9191"
       - "9292"
     environment:
-      DATALAND_EDC_WEB_URI: ${DATALAND_EDC_WEB_URI}
-      DATALAND_EDC_IDS_URI: ${DATALAND_EDC_IDS_URI}
-      TRUSTEE_WEB_URI: ${TRUSTEE_WEB_URI}
+      DATALAND_EDC_ASSET_ACCESS_URI: ${DATALAND_EDC_ASSET_ACCESS_URI}
       TRUSTEE_IDS_URI: ${TRUSTEE_IDS_URI}
```
Comment (fsoffel) [inline docker-compose.yml:158]: It seems to me that all of these are unused. What was your reason for removing exactly these three? Maybe also remove them in the .env.template.

File: docker-compose.yml [hunk 2]
```diff
@@ -237,6 +235,7 @@ services:
     volumes:
       - ./dataland-pgadmin/servers.json:/pgadmin4/servers.json:ro
       - ./dataland-pgadmin/pgpass.template:/pgpass.template:ro
+      - pgadmin-config:/var/lib/pgadmin
```
Comment (fsoffel) [inline docker-compose.yml:238]: Why is that added?

File: docker-compose.yml [hunk 3]
```diff
@@ -131,7 +131,7 @@ services:
   edc-dummyserver:
     restart: always
     profiles: ["testing", "development", "productionInMemory"]
-    image: ghcr.io/d-fine/datalandedc/edcdummyserver:0.2.8
+    image: ghcr.io/d-fine/datalandedc/edcdummyserver:0.2.30-SNAPSHOT
```
Comment (fsoffel) [inline docker-compose.yml:134]: Why this non-release update?

File: docker-compose.yml [hunk 4]
```diff
@@ -256,3 +257,7 @@ volumes:
     driver: local
   letsencrypt:
     driver: local
+  pgadmin_config:
+    driver: local
+  frontend-dev-node-modules:
+    driver: local
```
Comment (ppuetsch) [inline docker-compose.yml:263]: newline missing at end of file

Comment (fsoffel) [reply docker-compose.yml]: fixed

File: deployment/start_and_deploy_to_server.sh [hunk 5]
```diff
@@ -31,9 +31,11 @@ persistent_keycloak_backup_dir=/home/ubuntu/persistent_keycloak_backup
 keycloak_user_dir=$location/dataland-keycloak/users
 
 # shut down currently running dataland application and purge files on server
-ssh ubuntu@"$target_server_url" "(cd \"$location\" && sudo docker compose down) || true"
+ssh ubuntu@"$target_server_url" "(cd \"$location\" && sudo docker compose down --remove-orphans) || true"
 # make sure no remnants remain when docker-compose file changes
-ssh ubuntu@"$target_server_url" "docker kill $(docker ps -q); docker system prune --force; docker info"
+ssh ubuntu@"$target_server_url" "docker kill $(docker ps -q -a); docker rm $(docker ps -q -a); docker system prune --force; docker info"
+# delete pgadmin_config volume
+delete_docker_volume_if_existent_remotely "pgadmin_config" "$target_server_url" "$location"
```
Comment (ppuetsch) [inline deployment/start_and_deploy_to_server.sh:38]: heißt das volume nicht dataland_pgadmin_config?

Comment (fsoffel) [reply deployment/start_and_deploy_to_server.sh:38]: Ja, aber es sucht nach jedem volume mit dem substring im namen. Ich hab mich da an die notation wie es bei der backend db gemacht wurde gehalten.

Comment (SiegfriedSobkowiak) [reply deployment/start_and_deploy_to_server.sh:38]: es wird ja im Skript explizit die Liste der existierenden volumes durchsucht und das ist einfach grep also substringmatching

Comment (SiegfriedSobkowiak) [reply deployment/start_and_deploy_to_server.sh:38]: `search_volume () {
  volume_found=$(docker volume ls -q | grep "$1") || true
  echo "$volume_found"
}`
